File: 프로그래머스/1/42840. 모의고사/모의고사.py
def solution(answers):
    answer = []
    
    answerOne = [1, 2, 3, 4, 5]
    countOne = 0
    
    answerTwo = [2, 1, 2, 3, 2, 4, 2, 5]
    countTwo = 0
    
    answerThree = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5]
    countThree = 0
    
    for i in range(0, len(answers)):
        if (answers[i] == answerOne[i % 5]): countOne += 1;
        if (answers[i] == answerTwo[i % 8]): countTwo += 1;
        if (answers[i] == answerThree[i % 10]): countThree += 1; 
    
    temp = []
    temp.append(countOne);
    temp.append(countTwo);
    temp.append(countThree);
    maxScore = max(temp)
    
    for i in range(0,3):
        if (maxScore == temp[i]):
            answer.append(i+1)
        
    return answer

File: 프로그래머스/1/42840. 모의고사/모의고사.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> answers) {
    vector<int> answer;
    vector<int> first = {1, 2, 3, 4, 5};
    vector<int> second = {2, 1, 2, 3, 2, 4, 2, 5};
    vector<int> third = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    int one = 0;
    int two = 0;
    int three = 0;
    vector<int> temp(3, 0);
    for (int i = 0; i < answers.size(); i++) {
        if (answers[i] == first[i % 5]) temp[0]++;
        if (answers[i] == second[i % 8]) temp[1]++;
        if (answers[i] == third[i % 10]) temp[2]++;
    }
    
    int max = *max_element(temp.begin(), temp.end());
    
    for (int i = 0; i < temp.size(); i++) {
        if (temp[i] == max)
            answer.push_back(i+1);
    }
    
    return answer;
}